### PR TITLE
feat(header|p2p): add blacklisting to peerTracker

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview
Resolves https://github.com/celestiaorg/celestia-node/issues/1531

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by Wondertan — APPROVED** [review]

File: header/p2p/peer_tracker.go [hunk 1]
```diff
@@ -190,3 +194,25 @@ func (p *peerTracker) stop() {
...
+	// add peer to the blacklist, so we can't connect to it in the future.
+	err := p.connGater.BlockPeer(pID)
+	if err != nil {
+		log.Errorw("header/p2p: blocking peer failed", "err", err, "pID", pID)
+	}
+	// close connections to peer.
+	err = p.host.Network().ClosePeer(pID)
+	if err != nil {
+		log.Errorw("header/p2p: closing connection with peer failed", "err", err, "pID", pID)
+	}
```
Comment (renaynay) [inline header/p2p/peer_tracker.go:209]: small nit but i prefer to log err last, so format should be

some msg, pid, err

File: header/p2p/peer_tracker.go [hunk 2]
```diff
@@ -190,3 +194,25 @@ func (p *peerTracker) stop() {
...
+
+	log.Warnw("header/p2p: blocked peer", "pID", pID)
```
Comment (renaynay) [inline header/p2p/peer_tracker.go:211]: would be good to log reason too but we can do in separate PR

File: nodebuilder/header/module_test.go [hunk 3]
```diff
@@ -60,6 +61,9 @@ func TestConstructModule_ExchangeParams(t *testing.T) {
 			return datastore.NewMapDatastore()
 		}),
 		ConstructModule(node.Light, &cfg),
+		fx.Provide(func(b datastore.Batching) (*conngater.BasicConnectionGater, error) {
+			return conngater.NewBasicConnectionGater(b)
+		}),
```
Comment (renaynay) [inline nodebuilder/header/module_test.go:66]: we already provide a connection gater to fx in the p2p pkg, we don't need this.

**Review by renaynay — CHANGES_REQUESTED** [review]

**Review by renaynay — APPROVED** [review]

File: header/p2p/peer_tracker.go (hunk 2 above)
Comment (renaynay) [reply header/p2p/peer_tracker.go:211]: we can make an issue for it since it would break method, lets just keep retry to retry

Comment (vgonkivs) [reply header/p2p/peer_tracker.go:211]: ok, sure

File: header/p2p/peer_tracker.go (hunk 1 above)
Comment (vgonkivs) [reply header/p2p/peer_tracker.go:209]: Will do in a separate PR
